flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/Executing.java

```diff
@@ -281,9 +283,9 @@ void goToFailing(
     static final class FailureResult {
         @Nullable private final Duration backoffTime;
 
-        @Nullable private final Throwable failureCause;
+        private final Throwable failureCause;
 
-        private FailureResult(@Nullable Duration backoffTime, @Nullable Throwable failureCause) {
+        private FailureResult(Throwable failureCause, @Nullable Duration backoffTime) {
```

Comment:
Are you sure that this is safe? There are some cases where failure causes for tasks can be null (FLINK-21376), but I'm not sure if this applies here.

Reply:
Let me double check.

Reply:
I think it applies. I will add a safety net.